src/MALA.jl

```diff
@@ -0,0 +1,103 @@
+using ForwardDiff: gradient!
+using DiffResults: GradientResult, value, gradient
+
+struct MALA{D} <: MHSampler
+    proposal::D
+end
+
+
+MALAMH(d) = MALA(RandomWalkProposal(d))
+
+
+struct Transition_w∇{T<:Union{Vector, Real, NamedTuple}, L<:Real, G<:Union{Vector, Real, NamedTuple}}
+    params :: T
+    lp :: L
+    ∇ :: G
+end
+
+
+# Store the new draw, its log density and its gradient
+Transition_w∇(model::DensityModel, params) = Transition_w∇(params, ∂ℓπ∂θ(model, params)...)
+
+
+function propose(
+    rng::Random.AbstractRNG,
+    spl::MALA{<:Proposal},
+    model::DensityModel,
+    params_prev::Transition_w∇
+)
+    proposal = propose(rng, spl.proposal(params_prev.∇), model, params_prev.params)
+    return Transition_w∇(model, proposal)
+end
+
+
+function q(
+    spl::MALA{<:Proposal},
+    t::Transition_w∇,
+    t_cond::Transition_w∇
+)
+    return q(spl.proposal(-t_cond.∇), t.params, t_cond.params)
+end
+
+
+"""
+    ∂ℓπ∂θ(model::DensityModel, θ::T)
+
+Efficiently returns the value and gradient of the model
+"""
+function ∂ℓπ∂θ(model::DensityModel, params)
```

Comment:
I'm debating whether to change this to not use unicode characters, but since it's mostly an internal function I think I'm okay with leaving it as-is.

Reply:
I'm with you on the no-unicode style of the repo, so I'll get rid of it.